### DISCUSSION TITLE
[Curso 2811] Implementar a API completa?

### DISCUSSION THREAD

**Discussion by juliusczm** [balta-io/forum#678] [Tira Dúvidas] [answered]
Olá, Balta.

No curso 2811 vi que a implementação das "funcionalidades" da API do Blog foram limitadas ao User, Category e Post (Get, GetById e GetByCategory) a título de desenvolvimento nas aulas.

Vale a pena tentar fazer o restante (adicionar um post, adicionar tag a um post...), principalmente para os relacionamentos M:N (UserRole e PostTag), ou melhor seguir a carreira que isso será melhor abordado nos cursos subsequentes?

Vale a pena tentar fazer nessa API o desafio do Curso 2806? O Desafio era:

  - Cadastrar um usuário
  - Cadastrar um perfil (Role)
  - Vincular um usuário a um perfil
  - Listar os usuários (Nome, Email e Perfis separados por vírgula)
  - Cadastrar uma Categoria 
  - Cadastrar uma Tag
  - Cadastrar um Post
  - Vincular um post a uma tag
  - Listar todos os Posts com sua respectiva categoria
  - Listar os Posts de uma categoria
  - Listar os posts com suas tags (separados por vírgula)
  - Listar categorias com quantidades de posts
  - Listar Tags com quantidades de posts

**Comment by andrebaltieri** [comment, marked as answer]
@juliusczm vale sim!
